### PR TITLE
Update Test AV Script to do a quick scan for "unknown" modules loaded into Exchange

### PR DESCRIPTION
**Pull request by Canthv0**
**Issue:**
Need to check Exchange processes for 3rd party modules

**Reason:**
To test if AV etc. is loaded into Exchange.

**Fix:**
Pull all processes
compare process to list of known exchange process
compare loaded modules to whitelist of modules
Report if any found that are not in the white list

**Validation:**
* comment a module name from the whitelist in the code
* Run the script against and Exchange server

### CODE REVIEW

**Comment by dpaulson45** [comment]
@Canthv0 this is already being worked on in PR #1641

**Comment by Canthv0** [comment]
I noticed ... but I also see that, that has been in process for a few months now ... and we kinda need this now.

Plus, I didn't see that before I wrote the code for this ... so I figured this one is ready to go and I don't think needs major work.  We can make the call on which one to use or I will drop this one and wait on the other to be finished. <shrug> did the work so figured WTH ... :)

**Comment by Canthv0** [comment]
Duplicate closing as such

**Comment by Canthv0** [comment]
Failing on SpellCheck due to whitelist module names ... thoughts on how to fix this?

d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:321:30 - Unknown word (Protobuf)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:323:23 - Unknown word (Newtonsoft)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:333:23 - Unknown word (wvcore)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:334:23 - Unknown word (sccut)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:335:23 - Unknown word (sccfut)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:336:23 - Unknown word (sccfa)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:337:23 - Unknown word (sccfi)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:338:23 - Unknown word (sccch)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:339:23 - Unknown word (sccda)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:340:23 - Unknown word (sccfmt)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:341:23 - Unknown word (sccind)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:342:23 - Unknown word (sccca)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:343:23 - Unknown word (scclo)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:344:23 - Unknown word (SCCOLE)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:345:23 - Unknown word (SCCSD)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:346:23 - Unknown word (SCCXT)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:362:63 - Unknown word (fileversioninfo)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:362:79 - Unknown word (companyname)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:362:128 - Unknown word (fileversioninfo)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:362:144 - Unknown word (companyname)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:362:196 - Unknown word (companyname)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:366:67 - Unknown word (modulename)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:370:77 - Unknown word (processname)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:373:76 - Unknown word (processname)
d:\a\1\s\Diagnostics\AVTester\Test-ExchAVExclusions.ps1:373:107 - Unknown word (modulename)

**Comment by bill-long** [comment]
Best option might be to extract that list into a separate file and then exclude that file from the spell check in `.build\cpsell.json`. You can see a few other excluded files there.

File: Diagnostics/AVTester/Test-ExchAVExclusions.ps1 [hunk 1]
```diff
@@ -304,5 +304,85 @@ if ($BadFolderList.count -gt 0 -or $BadExtensionList.Count -gt 0 ) {
...
+
+Write-SimpleLogFile -string "Testing for AV loaded in processes" -name $LogFile -OutHost
+
+# Test Exchange Processes for unexpected modules
+$ProcessList = Get-ExchAVExclusionsProcess -ExchangePath $ExchangePath -MsiProductMinor ([byte]$serverExchangeInstallDirectory.MsiProductMinor)
+
+# Gather all processes on the computer
+$ServerProcess = Get-Process
+
+# Module white list
```
Comment (lusassl-msft) [inline Diagnostics/AVTester/Test-ExchAVExclusions.ps1:318]: Nit: let’s name this `AllowList` instead of using `WhiteList` term

**Review by lusassl-msft — CHANGES_REQUESTED** [review]

File: Diagnostics/AVTester/Test-ExchAVExclusions.ps1 (hunk 1 above)
Comment (Canthv0) [reply Diagnostics/AVTester/Test-ExchAVExclusions.ps1]: Updated WhiteList to AllowList

**Review by dpaulson45 — CHANGES_REQUESTED** [review]
@Canthv0 Let's also update the docs page for this script stating that we are checking for the modules loaded into the process as well.

**Review by dpaulson45 — APPROVED** [review]

**Review by lusassl-msft — APPROVED** [review]

**Pull request merged**